### PR TITLE
[JSONRPC]: fix - gcc8 build errors

### PR DESCRIPTION
**Pull request by moorthy-bs**
Signed-off-by: Moorthy Baskar <moorthy.baskaravenkatraman-sambamoorthy@linaro.org>

### CODE REVIEW

File: Source/websocket/JSONRPCLink.h [hunk 1]
```diff
@@ -1275,14 +1275,14 @@ namespace JSONRPC {
             uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method)
             {
                 Client::Subscribe<INBOUND, METHOD>(eventName, method);
```
Comment (HaseenaSainul) [inline Source/websocket/JSONRPCLink.h]: It will be good to remove this line and use only below one

File: Source/websocket/JSONRPCLink.h [hunk 2]
```diff
@@ -1275,14 +1275,14 @@ namespace JSONRPC {
...
-                return Client::Subscribe(waitTime, eventName);
+                return Client::Subscribe(waitTime, eventName, method);
```
Comment (HaseenaSainul) [inline Source/websocket/JSONRPCLink.h]: Please add template types also with function call, eg:, Client::Subscribe<INBOUND, METHOD>(waitTime, eventName, method);

File: Source/websocket/JSONRPCLink.h [hunk 3]
```diff
@@ -1275,14 +1275,14 @@ namespace JSONRPC {
...
             }
             template <typename INBOUND, typename METHOD, typename REALOBJECT>
             uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method, REALOBJECT* objectPtr)
             {
                 Client::Subscribe<INBOUND, METHOD, REALOBJECT>(eventName, method, objectPtr);
```
Comment (HaseenaSainul) [inline Source/websocket/JSONRPCLink.h]: Same as above comment

File: Source/websocket/JSONRPCLink.h [hunk 4]
```diff
@@ -1275,14 +1275,14 @@ namespace JSONRPC {
...
 
-                return Client::Subscribe(waitTime, eventName);
+                return Client::Subscribe(waitTime, eventName, method);
```
Comment (HaseenaSainul) [inline Source/websocket/JSONRPCLink.h]: Please add template types here as well, like: Client::Subscribe<INBOUND, METHOD, REALOBJECT>(waitTime, eventName, method, objectPtr);

File: Source/websocket/JSONRPCLink.h (hunk 2 above)
Comment (moorthy-bs) [reply Source/websocket/JSONRPCLink.h]: Done

File: Source/websocket/JSONRPCLink.h [hunk 5]
```diff
@@ -1274,15 +1274,12 @@ namespace JSONRPC {
             template <typename INBOUND, typename METHOD>
             uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method)
             {
-                Client::Subscribe<INBOUND, METHOD>(eventName, method);
-                return Client::Subscribe(waitTime, eventName);
+                return Client::Subscribe<INBOUND, METHOD>(eventName, method);
```
Comment (HaseenaSainul) [inline Source/websocket/JSONRPCLink.h]: please forward waitTime as well

File: Source/websocket/JSONRPCLink.h [hunk 6]
```diff
@@ -1274,15 +1274,12 @@ namespace JSONRPC {
...
             }
             template <typename INBOUND, typename METHOD, typename REALOBJECT>
             uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method, REALOBJECT* objectPtr)
             {
-                Client::Subscribe<INBOUND, METHOD, REALOBJECT>(eventName, method, objectPtr);
-
-                return Client::Subscribe(waitTime, eventName);
+                return Client::Subscribe<INBOUND, METHOD, REALOBJECT>(eventName, method, objectPtr);
```
Comment (HaseenaSainul) [inline Source/websocket/JSONRPCLink.h]: please forward waitTime here as well

**Pull request merged**